src/MQTT.h

```diff
@@ -153,41 +205,296 @@ void mqtt_callback(char *topic, byte *data, unsigned int length) {
+ *
+ * This function generates a number device configuration for Home Assistant MQTT discovery. It creates a `DiscoveryObject` containing the necessary information for Home Assistant to discover and control the number device.
+ *
+ * @param name The name of the number device (used in MQTT topics)
+ * @param displayName The display name of the number device (shown in Home Assistant)
+ * @param min_value The minimum value allowed for the number device
+ * @param max_value The maximum value allowed for the number device
+ * @param steps_value The step value for incrementing/decrementing the number device value
+ * @param unit_of_measurement The unit of measurement for the number device (default: "°C")
+ * @param ui_mode Control how the number should be displayed in the UI
+ * @return A `DiscoveryObject` containing the number device configuration
+ */
+DiscoveryObject GenerateNumberDevice(String name, String displayName, int min_value, int max_value, float steps_value, String unit_of_measurement = "°C", String ui_mode = "box") {
+  String mqtt_topic = String(mqtt_topic_prefix) + String(hostname);
+  DiscoveryObject number_device;
+  String unique_id = "clevercoffee-" + String(hostname);
+
+  String NumberDiscoveryTopic = String(MQTT_HASSIO_DISCOVERY_PREFIX) + "/number/";
+  number_device.discovery_topic = NumberDiscoveryTopic + unique_id + "-"+ name +"" +"/config";
+
+  DynamicJsonDocument DeviceMapDoc(1024);
+  char DeviceMapBuffer[256];
+  DeviceMapDoc["identifiers"] = String(hostname);
+  DeviceMapDoc["manufacturer"] = "CleverCoffee";
+  DeviceMapDoc["model"] = getMachineName(machine);
+  DeviceMapDoc["name"] = String(hostname);
+
+  DynamicJsonDocument numberConfigDoc(512);
+  numberConfigDoc["name"] = displayName;
+  numberConfigDoc["command_topic"] = mqtt_topic + "/"+ name + "/set";
+  numberConfigDoc["state_topic"] = mqtt_topic + "/"+ name;
+  numberConfigDoc["unique_id"] = unique_id + "-"+name;
+  numberConfigDoc["min"] = min_value;  // Minimum value
+  numberConfigDoc["max"] = max_value;  // Maximum value
+  numberConfigDoc["step"] = String(steps_value,2);  // Steps value
+  numberConfigDoc["unit_of_measurement"] = unit_of_measurement;
+  numberConfigDoc["mode"] = ui_mode;
+  numberConfigDoc["payload_available"] = "online";
+  numberConfigDoc["payload_not_available"] = "offline";
+  numberConfigDoc["availability_topic"] = mqtt_topic+"/status";
+
+  JsonObject numberDeviceField = numberConfigDoc.createNestedObject("device");
+  for (JsonPair keyValue : DeviceMapDoc.as<JsonObject>()) {
+    numberDeviceField[keyValue.key()] = keyValue.value();
+  }
+
+  String numberConfigDocBuffer;
+  serializeJson(numberConfigDoc, numberConfigDocBuffer);
+
+
+  number_device.payload_json = numberConfigDocBuffer;
+  return number_device;
+}
+
+/**
+ * @brief Send MQTT Homeassistant Discovery Messages
+ * @return 0 if successful, MQTT connection error code if failed to send messages
+ */
+int sendHASSIODiscoveryMsg() {
```

Comment:
I suggest improving the texts
` DiscoveryObject brewSetpoint = GenerateNumberDevice("brewSetpoint", "Brew setpoint", BREW_SETPOINT_MIN, BREW_SETPOINT_MAX, 0.1, "°C");
  DiscoveryObject steamSetPoint = GenerateNumberDevice("steamSetpoint", "Steam setpoint", STEAM_SETPOINT_MIN, STEAM_SETPOINT_MAX, 0.1, "°C");
  DiscoveryObject brewTempOffset = GenerateNumberDevice("brewTempOffset", "Brew Temp. Offset", BREW_TEMP_OFFSET_MIN, BREW_TEMP_OFFSET_MAX, 0.1, "°C");
  DiscoveryObject brewPidDelay = GenerateNumberDevice("brewPidDelay", "Brew Pid Delay", BREW_PID_DELAY_MIN, BREW_PID_DELAY_MAX, 0.1, "");
  DiscoveryObject startKp = GenerateNumberDevice("startKp", "Start kP", PID_KP_START_MIN, PID_KP_START_MAX, 0.1, "");
  DiscoveryObject startTn = GenerateNumberDevice("startTn", "Start Tn", PID_TN_START_MIN, PID_TN_START_MAX, 0.1, "");
  DiscoveryObject steamKp = GenerateNumberDevice("steamKp", "Start Kp", PID_KP_STEAM_MIN, PID_KP_STEAM_MAX, 0.1, "");
  DiscoveryObject aggKp = GenerateNumberDevice("aggKp", "aggKp", PID_KP_REGULAR_MIN, PID_KP_REGULAR_MAX, 0.1, "");
  DiscoveryObject aggTn = GenerateNumberDevice("aggTn", "aggTn", PID_TN_REGULAR_MIN, PID_TN_REGULAR_MAX, 0.1, "");
  DiscoveryObject aggTv = GenerateNumberDevice("aggTv", "aggTv", PID_TV_REGULAR_MIN, PID_TV_REGULAR_MAX, 0.1, "");
  DiscoveryObject aggIMax = GenerateNumberDevice("aggIMax", "aggIMax", PID_I_MAX_REGULAR_MIN, PID_I_MAX_REGULAR_MAX, 0.1, "");
  DiscoveryObject brewtime = GenerateNumberDevice("brewtime", "Brew time", BREW_TIME_MIN, BREW_TIME_MAX, 0.1, "s");
// Sensor Devices
  DiscoveryObject actual_temperature = GenerateSensorDevice("temperature", "Boiler Temperature");
  DiscoveryObject heaterPower = GenerateSensorDevice("heaterPower", "Heater Power", "ms", "power_factor");
// Switch Devices
  DiscoveryObject pidOn = GenerateSwitchDevice("pidON", "Use PID");
  DiscoveryObject steamON = GenerateSwitchDevice("steamON", "Steam");
  DiscoveryObject backflushON = GenerateSwitchDevice("backflushON", "Backflush");
  DiscoveryObject startUsePonM = GenerateSwitchDevice("startUsePonM", "Use PonM");
  `